### DISCUSSION TITLE
Problema duckdns in wi-fi

### DISCUSSION THREAD

**Discussion by katanza12** [indomus/forum#701] [HUB: Home Assistant]
Ciao a tutti.
Ho un problema che non so bene nè come risolvere nè come catalogare. Ho un Pi4 con HA Core installato ed aggiornato, caddy come reverse proxy e duckdns come dyndns. Fino a 2 settimane fa funzionava tutto perfettamente. Tornato dalle vacanze invece, non riesco più ad accedere ad HA nel senso che il browser (qualunque sia, anche in modalità anonima) va in timeout. 
Duckdns è allineato, HA è all'ultima versione e non da errori, caddy funziona perchè ho altre porte mappate.

La cosa strana è che succede solo dal pc in wifi:
- via ip interno funziona, quindi l'istallazione non è corrotta o qualcosa del genere
- dal telefono funziona solo se in 4G
- se collego il pc al telefono in hotspot funziona
- tutto il resto di internet non mi da problemi
Ho già riavviato il raspberry, il modem, il pc...

Qualche idea?

Grazie

**Comment by snakuzzo** [comment]
Dalla tua wifi, se punti un altro servizio esposto (non HA) hai lo stesso problema?
Lato caddy abilita i log per i servizi configurati 3 vedi se almeno lì arrivano le chiamate.

A pelle mi verrebbe da dire... NAT loopback

**Comment by katanza12** [comment]
Se da wifi punto all'indirizzo di duckdns con un'altra porta su cui ho mappato tasmoadmin (su caddy), lo raggiungo senza problemi.

Ora provo ad attivare i log, ma quello che mi lascia un po' perplesso è che non ho modificato niente (se non gli aggiornamenti) tra quando funzionava ed ora!

**Reply by snakuzzo** [reply]
a questo punto mi viene solo in mente che possa essere un problema di portforwarding

**Reply by katanza12** [reply]
Ho controllato: sul modem la 443 (usata per HA) e la 9542 (usata per TasmoAdmin) sono configurate allo stesso modo

**Reply by katanza12** [reply]
E poi, se così fosse dovrei avere anche lo stesso problema indipendentemente dalla connessione (wifi o 4g)

**Reply by snakuzzo** [reply]
per questo all'inizio ti parlavo di nat loopback...però mi sembra troppo strano.
E così al volo non saprei cosa consigliarti

**Reply by Newreader** [reply]
Il problema, secondo il mio modesto parere, sta nel router che viene utilizzato per la connessione. Il wifi non c'entra perché con l'hotspot, creato con lo smartphone, il wifi funziona.
A titolo di curiosità hai anche provato a cambiare browser?
Hai per caso salvato un'immagine del sistema fatta quando funzionava il tutto?

**Reply by katanza12** [reply]
Ho provato chrome, firefox, sia normali, sia in anonimo, sia dopo aver svuotato la cache. 
L'immagine salvata... ma di che cosa? HomeAssistant? Del router (Frtizbox)?

**Reply by Newreader** [reply]
Tu utilizzi HA Core o Hassio? Inoltre visto che hai un Fritz!Box la funzione di associazione dell'IP al tuo FQDN la fai assolvere del modem/router stesso? Perché, riprendendo le parole testuali della guida "Installare e configurare Caddy con Docker su Raspberry Pi OS", "Se e quando un modem è in grado di provvedere a tale funzione, è sempre buona regola lasciare che sia lui ad aggiornare tale associazione, e non il Raspberry o altri host della propria rete."
Per immagine intendo una "istantanea" del sistema. In ultima analisi anche una copia del supporto (MicroSd, ad esempio) su cui gira tutto il sistema, fatta ovviamente con un tools che supporta il disk imaging.

**Reply by katanza12** [reply]
Uso HA Core e l'associazione dell'ip non la faccio fare al modem perchè non ha la configurazione per duckdns già preimpostata ed in passato ho avuto dei disallineamenti quando ho fatto tale configurazione manualmente. Ad ogni modo, l'ip è allineato correttamente, tant'è che se punto a https://xxxx.duckdns.org:9542 mi risponde TasmoAdmin installato sullo stesso raspberry. Invece sulla 443, dove dovrebbe rispondere HA, non va!

**Reply by Newreader** [reply]
Ok. La mia esperienza con il Fritz!Box è sempre stata positiva. Mai avuto nessun problema. Comunque .... Dalla tua risposta deduco che non hai a disposizione un'immagine del sistema. Hai provato a verificare cosa accade se cerchi di accedere attraverso l'app HA companion?

**Reply by katanza12** [reply]
Scusa, avevo dimenticato di rispondere a quella parte. Un'immagine potrei anche averla, ma non ho la certezza che sia buona perchè ho il backup automatico settimanale e visto che la scorsa settimana ero in vacanza e non ho mai aperto l'app, non mi sono accorto del problema. 
Con l'app su iOS (non) funziona allo stesso modo: col wifi solo tramite l'ip, in 4G funziona l'url duckdns

**Reply by Newreader** [reply]
Inoltre se puoi accedere in SSH magari attraverso Putty prova a dare il comando "sudo docker ps" così visualizzi la lista dei contenitori attivi  comprese le porte ad essi collegate (fra cui la 443), ovvero dovresti avere per PORTS una risposta del tipo:
 80/tcp, 0.0.0.0:443->443/tcp, 2019/tcp, 0.0.0.0:9000->9000/tcp

**Reply by Newreader** [reply]
Puoi quindi fare un backup del tuo attuale sistema e rimontare la copia che hai a disposizione e di cui non sei sicuro. Se le cose non dovessero funzionare puoi reinstallare l'attuale di cui, come scritto prima, hai preventivamente provveduto a fare il backup.

**Reply by katanza12** [reply]
Sembra che la 443 non sia rimappata su Caddy.

Questo è quello che c'è sul CaddyFile:
```
#HomeAssistant
https://xxx.duckdns.org:443 {
  import https_header
  reverse_proxy http://192.168.1.6:8123
}
```
Questi i parametri di lancio di Caddy:
`docker run --init -d --name="caddy" -e "TZ=Europe/Rome" -v /home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile -v /home/pi/caddy/data:/data --restart always -p 9000:9000 -p 9542:9542 caddy:latest`

E questo è il risultato di "sudo docker ps" relativo alle porte di Caddy:
` 80/tcp, 443/tcp, 0.0.0.0:9000->9000/tcp, 2019/tcp, 0.0.0.0:9542->9542/tcp`

**Reply by Newreader** [reply]
A me sembra che nel comando per lanciare Caddy manchi tra i parametri proprio il riferimento alla porta 443 ovvero  -p 443:443 dopo --restart always

**Reply by snakuzzo** [reply]
La 443 non è necessaria se non hai intenzione di usare la challenge tls-apln-01 per il rilascio del certificato...stesso dicasi per la challenge http-01.

Però se la usi per i servizi...devi esporla :D

**Reply by katanza12** [reply]
Ho provato ad aggiungerla, sia con "-p 443:443", sia "-p 443:8123" ma il risultato non è cambiato. E poi la domanda di base rimane: perchè un mese fa funzionava?
Quello che ho scoperto nel frattempo è che il FritzBox si è aggiornato (sto cercando il change completo), non vorrei che abbiano bloccato il loopback come diceva Snakuzzo all'inizio...

**Reply by snakuzzo** [reply]
Io ho un fritzbox e uso caddy da mesi...e non è cambiato niente. Tra le altre cose se funzionano altri servizi e solo ha ti dà problemi, non credo sia quello il problema.

Per il discorso porte però non devi andare facendo prove a caso. Una è la porta con la quale mappi all'esterno la porta interna, l'altra è appunto la porta interna. Se mappi la 443 sulla 8123...non funzionerà mai, perché atterrerai sul container di caddy che però non espone la 8123... non so se mi sono spiegato.

Per come hai scritto il Caddyfile mapperai la 443 sulla 443 interna di caddy che a sua volta ti proxerà HA che risponde (in questo caso) con l'ip host sulla 8123

**Reply by snakuzzo** [reply]
Comunque se non riesci...stasera sul tardi se vuoi posso darti una mano da remoto...

**Reply by snakuzzo** [reply]
Ricordati anche di configurare i trusted proxy sulla config di HA

**Reply by Newreader** [reply]
Anch'io ho un FRitz!Box (sul quale ho abilitato il DynDNS) e uso Caddy (su Docker) dall'inizio dell'anno e non ho mai avuto problemi (né con gli aggiornamenti né con altro).
Solo per curiosità, tu per i certificati crittografici quale servizio usi?
L'aiuto di snakuzzo è sicuramente prezioso e insostituibile, ma qualche altra indagine cerca di farla.
Comunque potresti, se vuoi, ripristinare il vecchio backup (altrimenti che lo fai a fare), dopo aver salvato la tua attuale immagine,  che sicuramente non ti fa capire la causa dei tuoi problemi ma potrebbe sortire l'effetto disiderato di risolverli.

**Reply by snakuzzo** [reply]
...non fai il backup per queste piccole cose di configurazione. Il backip magari ti serve se perdi tutto, ma per ste cose ci si mette e si risolve

**Reply by Newreader** [reply]
Hai ragione ma la mia era una "ultima ratio" ( a costo zero)
Per sapere l'indirizzo IP del reverse proxy da inserire nel trusted_proxies della config di HA, potrebbe esserti utile lanciare il comando:
sudo docker network inspect bridge
che tra le altre ti fornisce questa informazione

**Reply by katanza12** [reply]
Scusate, ieri pomeriggio sono stato preso dal lavoro e poi non mi sono più collegato. Comunque grazie mille per il supporto che mi state dando!!

@snakuzzo sono d'accordo con te sulle prove "a caso", ma era una sorta di tentativo disperato! Ho pensato che magari non avevo chiaro il funzionamento di quel parametro e visto che mi ci voleva 1 minuto per fare il tentativo, ho fatto che provarci!

Questo è il risultato dell'inspect bridge:
```
[
    {
        "Name": "bridge",
        "Id": "93d038ced8211bcda110409595cd5473924e724bf1008a2bda581349c68b4a54",
        "Created": "2021-09-06T10:50:46.976494924+01:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.17.0.0/16",
                    "Gateway": "172.17.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "7ed353f28d8b28759098f85ff730c82636ccd53c27b92b7becd74a05a0779778": {
                "Name": "caddy",
                "EndpointID": "4e99b0b7bc62e99dadf50016f6954fd4b4c5694fa702c6c57a36d1462190b375",
                "MacAddress": "02:42:ac:11:00:02",
                "IPv4Address": "172.17.0.2/16",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.bridge.default_bridge": "true",
            "com.docker.network.bridge.enable_icc": "true",
            "com.docker.network.bridge.enable_ip_masquerade": "true",
            "com.docker.network.bridge.host_binding_ipv4": "0.0.0.0",
            "com.docker.network.bridge.name": "docker0",
            "com.docker.network.driver.mtu": "1500"
        },
        "Labels": {}
    }
]

```
Quindi, se lo leggo correttamente, l'ip dovrebbe essere "172.17.0.2/16"
Questo è quello che ho nel configuration.yaml per quanto riguarda i trusted proxy:
```
http:
  use_x_forwarded_for: true
  trusted_proxies: 
    - 127.0.0.1
    - ::1
    - 172.17.0.2
```
Dovrebbe essere corretto, o sbaglio?

**Reply by snakuzzo** [reply]
In realtà ti basta solo l'ip di Caddy `172.17.0.2`

**Reply by katanza12** [reply]
Ma gli altri danno fastidio o sono al più inutili?

**Reply by snakuzzo** [reply]
Se non hai errori quando fai partire HA, vuol dire che sono solo superflui :)

**Reply by Newreader** [reply]
Se lanci il comando (attraverso SSH) "sudo route -n"  visualizzarai la tabella di routing dove tra gli altri viene elencato come destinazione anche l'indirizzo del tuo docker che dovrebbe essere 172.17.0.0, il che suggerirebbe di indicare come trusted_proxies: 172.17.0.0/16 (dove /16 sta ad indicare il blocco di un certo numero di indirizzi). In questo caso si ritornerebbe alla guida InDomus che giustamente parla di indicare come trusted_proxies proprio 172.17.0.0/16 (questo implica che stiamo utilizzando docker e che Caddy e HA girano sullo stesso computer.
Comunque tornando a HA Companion, tu hai scritto che non ti fa collegare a HA, questo però ti dovrebbe permettere di impostare l'indirizzo di collegamento a mano, a questo punto prova ad inserire il seguente collegamento:
https://tuodominio.duckdns.org:443 (visto che hai mappato questa porta nel tuo CaddyFile))
Poi sarebbe utile come ho scritto precedentemente sapere come ottieni i certificati crittografici (Let's Encrypt o altro)

**Reply by katanza12** [reply]
L'ip del docker è effettivamente 172.17.0.0 ed anche mettendo 172.17.0.0/16 nei trusted_proxies la situazione non cambia.

Sull'app (come sul browser del telefono, come sul browser del pc) non mi fa collegare anche mettendo l'indirizzo completo di porta.

I certificati li ho fatti con "Dehydrated" seguendo [questa ](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-installato-su-raspberry-raspbian/) guida di indomus. Ho controllato e scadono a Dicembre, quindi dovrebbero essere a posto

**Reply by snakuzzo** [reply]
Ok, io però vorrei dare un'occhiata in prima persona se possibile.

Questa cosa di mischiare le cose in questo modo, non mi piace affatto.
Se vuoi usare docker, ti consiglierei di fare tutto su Docker, magari con docker-compose così ti riesce tutto più facilmente.
Come reverse proxy inoltre ti consiglio di utilizzare Caddy che si preoccupa direttamente lui di gestire la parte https con richiesta e rinnovo dei certificati in completa autonomia.

**Reply by Newreader** [reply]
Dire che snakuzzo ha ragione è pleonastico. Mischiare le cose non va mai bene.
Tu stai usando Caddy come reverse proxy e lo hai istanziato su docker (così perlomeno hai scritto), Per la sua installazione hai seguito la guida InDomus "Installare e configurare Caddy con Docker su Raspberry Pi OS"?
Ho il dubbio, forse sbagliato, che ci sia qualcosa che non va nei certificati crittografici e nella loro autenticazione.
La vecchia configurazione esplicitata nel file configuration.yaml, richiamata dalla guida da te menzionata "Collegarsi da remoto a Home Assistant Core installato su Raspberry Pi OS", ovvero:
```
http:
   ssl_certificate:
   ssl_key:
```
deve essere sostituita dalla nuova  così come evendenziata nella richiamata guida "Installare e configurare Caddy con Docker su Raspberry Pi OS", perchè sara Caddy che si farà carico di richiedere i nuovi certificati e rinnovarli a tempo debito.

**Reply by katanza12** [reply]
Non sono un esperto di linux e l'unica guida che avevo trovato per il reverse proxy era proprio quella indicata "Installare e configurare Caddy con Docker su Raspberry Pi OS". 
Il dubbio però è sempre lo stesso: perchè prima funzionava?

**Reply by katanza12** [reply]
E perchè n 4G funziona mentre in wifi no?

**Reply by katanza12** [reply]
Ho controllato meglio il discorso certificati e Dehydrated era quello che usavo prima di configurare Caddy. Adesso effettivamente è lui ad occuparsi anche dei certificati

**Reply by snakuzzo** [reply]
Io ho terminato le idee...
se ti va possiamo vedere insieme da remoto.

**Reply by katanza12** [reply]
Volentieri, dobbiamo solo trovare un momento buono per entrambi!

**Reply by snakuzzo** [reply]
La sera sul tardi...tipo 23.30

**Reply by katanza12** [reply]
Ci aggiorniamo più tardi e vediamo se riesco

**Reply by Newreader** [reply]
Bene. Se volete fatemi sapere l'origine del mancato collegamento, quando lo risolverete.
Come ultimo apporto consiglio di cambiare la porta da 443 a 9000 come riportato nella guida InDomus (il problema potrebbe essere lì), ovvero riprendendo dalla citata guida "Installare e configurare Caddy con Docker su Raspberry Pi OS":
Le cose da fare sono:
modificare coerentemente il “Caddyfile” sostituendo la porta 443 con la porta 9000;
configurare il port forwarding sul modem/router aprendo come porta interna ed esterna la 9000 con riferimento all'IP del Raspberry;
riavviare il container Docker di Caddy con i comandi ivi descritti:
docker container stop caddy
docker container rm caddy
docker run --init -d --name="caddy" -e "TZ=Europe/Rome" -v /home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile -v /home/pi/caddy/data:/data --restart always -p 443:443 -p 9000:9000 caddy:latest.

Infine verificare con il comando:
sudo docker ps
la risposta corretta del sistema

Inoltre, e questo lo si può fare preventivamente, se è stata seguita la guida menzionata qual è la risposta al comando:
sudo docker logs duckdns

**Reply by katanza12** [reply]
Ho provato a usare la porta 9000 ed effettivamente funziona, quindi non è un problema di HA. Quello che però volevo io era NON usare una porta specifica per l'indirizzo esterno, per quello volevo usare la 443.
Questa prova dovrebbe anche escludere problemi col certificato, giusto?

**Reply by snakuzzo** [reply]
Se ti funziona, vuol dire che non hai problemi con il certificato

**Reply by Newreader** [reply]
Bene! Mi sembra che l'obiettivo sia stato raggiunto.

**Reply by katanza12** [reply]
@snakuzzo se facciamo tutto senza audio (il resto della famiglia dorme) ci sono!

**Reply by snakuzzo** [reply]
Ciao...scusa @katanza12 ma sono stato impegnato.
Comunque ti conviene contattarmi su Telegram

**Reply by Newreader** [reply]
Nell'attesa di futuri collegamenti con snakuzzo puoi lanciare (SSH) il comando:
sudo netstat -pant
che ti mostrerà, tra l'atro, i local adderss con le relative porte che sono in ascolto e gli indirizzi esterni collegati, unitamente al nome dei programmi che li stanno utilizzando.
In questo modo vedrai elencate sia la porta 443 che la 9000 (se la stai utilizzando), con tutte le informazioni di cui sopra.

**Reply by katanza12** [reply]
Non so però come leggerlo...

**Reply by Newreader** [reply]
Poiché sono fermamente convinto che la configurazione con la porta 443 funziona (ovvero senza tirare il ballo la porta 9000) ho fatto la modifica del caddyfile con tutto ciò che ne consegue.
Ti riporto passo passo quello che ho fatto:
1. Modifica caddyfile in questo modo:
```
 (https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}

https://areaprivata.duckdns.org {
  import https_header
  reverse_proxy http://IP_DELLA_ISTANZA_HOME_ASSISTANT:8123
}

```
2. lanciati i seguenti comandi 
    sudo docker container stop caddy
    sudo docker container rm caddy
    sudo docker run --init -d --name="caddy" -e "TZ=Europe/Rome" -v /home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile -v /home/pi/caddy/data:/data --restart always -p 443:443 caddy:latest
3. sudo docker ps per verificare l'effettiva creazione e il collegamento con la porta 443
4.  sudo systemctl restart home-assistant@homeassistant
Al riavvio tutto ha funzionato come dovuto, ovvero per il collegamento  dall'esterno ho impostato il seguente indirizzo:
https://miodominio.duckdns.org         (non è necessario specificare la porta perché per Caddy la 443 è di default)
per il collegamento dall'interno della rete:
http://miodominio.duckdns.org:8123
Prova queste modifiche e fammi cortesemente sapere

**Reply by katanza12** [reply]
In realtà non ho dovuto fare nessuna modifica perchè è esattamente la configurazione che ho io (per scrupolo ho ricontrollato tutto) e che fino a 2 settimane fa funzionava perfettamente. 
Quello che secondo me non stai considerando è che dall'esterno funziona: se vado in 4g (o dal cellulare o con lo stesso in hotspot per il pc) risponde perfettamente. Non va solo in wifi! Quindi deve centrare qualcosa il fatto che dalla rete interna (wifi) io vada fuori (duckdns) per poi ritornare dentro (Caddy) allo stesso punto...

**Reply by Newreader** [reply]
Se con la porta 9000 funziona non è problema di wifi, ma di porta 443
Il  comando
sudo netstat -pant 
che risposta ti da?
oppure per semplificare il comando 
sudo docker ps
ti restituisce per caddy le seguenti porte:
80/tcp, 2019/tcp, 0.0.0.0:443->443/tcp
con  particolare riferimento alla 443?
È importante che ti restituisca 0.0.0.0:443->443/tcp, altrimenti non andiamo da nessuna parte.
Puoi verificare e pubblicare il risultato?

**Reply by katanza12** [reply]
Questo è quello che mi restituisce.
`80/tcp, 0.0.0.0:443->443/tcp, 0.0.0.0:9000->9000/tcp, 2019/tcp, 0.0.0.0:9542->9542/tcp   caddy`

Secondo me il problema è di wifi sulla 443, perchè la stessa porta e la stessa configurazione in 4G funziona

**Reply by Newreader** [reply]
Puoi provare a riconfigurare caddy solo con la porta 443 eliminando (per il momento) la 9000 e la 9542, modificando, se necessario, il CaddyFile?

**Reply by snakuzzo** [reply]
@katanza12 se vuoi possiamo provare un attimo da remoto... però mi devi contattare su telegram

**Reply by snakuzzo** [reply]
ma perché hai tutto mischiato ?

HA come servizio
Caddy su docker

Così fai solo casino dal mio punto di vista

**Reply by snakuzzo** [reply]
...e poi consiglio di usare docker-compose ... che ti semplifica molto la vita

**Reply by Newreader** [reply]
Alla fine della vostra indagine quale problema si era manifestato?

**Reply by snakuzzo** [reply]
Nessuna indagine...non mi ha contattato

**Reply by katanza12** [reply]
@snakuzzo scusa, ma ieri sera l'operazione di messa a nanna ha richiesto più tempo del solito.
Anche a me non piace avere tutto mischiato, ma esiste una guida per istallare Caddy NON su docker? Nel caso provvedo subito alla sostituzione.

@Newreader ho provato solo con la 443 sia nel Caddyfile, sia nei parametri di lancio ma non è cambiato nulla

**Reply by snakuzzo** [reply]
Non è che ci sia tanto da fare se vuoi installare tutto fuori docker...
Cerca su google "install caddy" ... primo risultato ... https://caddyserver.com/docs/install#debian-ubuntu-raspbian
Poi modifichi il Caddyfile come già stai facendo ora.

Ma ripeto, consiglio da amico dopo aver provato "la qualunque", se riesci a fissarti in testa i concetti principali di Docker relativi soprattutto a volumi e network, con docker-compose è un attimo a tirar su tutto quanto

**Reply by Newreader** [reply]
Tanto per indagare la porta 443, prova a dare questo comando e pubblica la risposta:
sudo netstat -pant | grep 443

Se installi Caddy come servizio ricordati di modificare il blocco http nel file di configurazione, ovvero indicare come trusted_proxies
127.0.0.1, se Caddy o il reverse proxy è installato sulla stesso computer di Home Assistant come servizio

**Reply by katanza12** [reply]
@Newreader  questo è il risultato:
```
tcp       25      0 192.168.1.6:56474       18.156.68.22:443        CLOSE_WAIT  28465/python3.8
tcp        0      0 192.168.1.6:39998       52.84.148.28:443        ESTABLISHED 28465/python3.8
tcp        0      0 127.0.0.1:43672         127.0.0.1:44435         ESTABLISHED 1460/python3
tcp       32      0 192.168.1.6:57424       54.77.37.248:443        CLOSE_WAIT  28465/python3.8
tcp        0      0 192.168.1.6:35002       151.101.128.223:443     TIME_WAIT   -
tcp        0      0 192.168.1.6:42530       149.154.167.220:443     ESTABLISHED 28465/python3.8
tcp        0      0 192.168.1.6:35138       18.156.61.1:443         ESTABLISHED 28465/python3.8
tcp        0      0 192.168.1.6:41724       54.239.27.11:443        ESTABLISHED 28465/python3.8
tcp        0      1 192.168.1.6:49206       51.183.145.195:443      SYN_SENT    28466/python3
tcp        0      0 127.0.0.1:44435         127.0.0.1:43672         ESTABLISHED 1460/python3
tcp       25      0 192.168.1.6:35534       54.188.51.105:443       CLOSE_WAIT  28465/python3.8
tcp        0      0 192.168.1.6:45540       34.231.216.212:443      TIME_WAIT   -
```
192.168.1.6 è l'IP del raspberry

**Comment by Newreader** [comment]
Come tu stesso puoi constatare, non hai la porta 443 in LISTEN, ovvero in ascolto, e soprattutto non c'è alcun riferimento al docker-proxy. In altre parole dovresti avere una riga del tipo:
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      1323/docker-proxy
affinché tutto proceda bene.
Fermo restando le cose così non funzionerà 443.
Se tu hai attivato la porta 9000 come da mio suggerimento iniziale e dai il comando:
sudo netstat -pant | grep 9000
posto che tu riesca a collegarti in remoto, come hai scritto, sicuramente troverai la riga (eccetto il PID):
tcp        0      0 0.0.0.0:9000            0.0.0.0:*               LISTEN      1289/docker-proxy
come correttamente deve essere per poter stabilire il collegamento.
Quindi in ultima analisi la porta 443 non è stata correttamente aperta verso il docker-proxy.
Al tuo posto rifarei tutta la procedura descritta nella guida InDomus "Installare e configurare Caddy con Docker su Raspberry Pi OS", posto che per altre installazioni il docker funzioni

**Reply by katanza12** [reply]
Rieccomi dopo un po' di giorni e qualche prova.
Ho rimosso il docker ed ho installato Caddy direttamente su raspbian. La situazione però non è cambiata: dall'esterno funziona, dal wifi no.
Ora se faccio il netstat però mi trovo questa riga:
`tcp6       0      0 :::443                  :::*                    LISTEN      11079/caddy`

**Reply by katanza12** [reply]
Ho trovato questo documento sul FritzBox che sembra esattamente il mio problema:
https://it.avm.de/assistenza/banca-dati-informativa/dok/FRITZ-Box-7330-int/663_La-risoluzione-DNS-di-indirizzi-IP-privati-non-e-possibile/

Ma la soluzione proposta non cambia nulla!

**Reply by Newreader** [reply]
Secondo me c'è qualcosa che sfugge magari perché non evidenziato nelle precedenti discussioni.
Un fatto è certo che con Caddy su Docker  utilizzando la porta 9000 tutto funzionava. Questo elimina un po' di problematiche sollevate precedentemente.
Sarebbe da indagare più a fondo cosa "passa" attraverso la porta 443.
A questo punto il consiglio che posso darti, onde evitare ulteriori perdite di tempo, è quello di raccogliere l'invito che ti è stato fatto da @snakuzzo "se vuoi possiamo provare un attimo da remoto... però mi devi contattare su telegram",

**Comment by katanza12** [comment]
Ciao a tutti. 
Mi ripropongo con questo argomento perchè ieri credo di aver fatto un passo avanti. Ricapitolando il problema: non riesco a collegarmi con duckdns se sono collegato alla rete interna, ma ci riesco da fuori. Da dentro riesco solo con l'ip (quindi HA funziona). HA Core gira su rasp Pi4, collegato ad un Fritz!Box che fa da router, collegato al modem Sky (che fa solo da modem e inoltra al Fritz tutte le chiamate su tutte le porte). 
Utilizzo caddy che fa da reverse proxy e gira "https://xxxx.duckdns.org:443" su "http://192.168.1.6:8123".

Il problema potrebbe essere che Sky non permette il nat loopback (che impedisce cioè di entrare dallo stesso "canale" da cui si è usciti), ma nessuno di quelli dell'assistenza con cui ho parlato è riuscito a rispondermi. Però, ieri ho notato una cosa: ho messo nel Caddyfile un puntamento semplice:
```
https://xxxx.duckdns.org:9000 {
  respond "Saluti dal Casa"
  file_server
  log {
        output file /var/log/access.log
  }
}
```
e funziona sia da dentro, sia da fuori. Quindi, in teoria e se ho capito bene come funziona tutto il giro, riesco a rientrare fino a Caddy, ma poi non fa l'ultimo step reindirizzando verso HA. Corretto?
In questo caso il problema sta o nella configurazione di Caddy o su quella di HA....

Qualcuno ha qualche idea in merito?

Grazie a tutti

**Reply by katanza12** [reply]
Aggiungo un pezzo, che dovrebbe definitivamente indicare che il problema sta su Caddy: nel mio Caddyfile ho modificato la entry di HA provando una porta diversa dalla 443 e fuziona! Qualcuno sa se Caddy ha qualche limitazione con la 443?

**Reply by snakuzzo** [reply]
Non c'è alcuna limitazione sulle porte. La 443 viene anche usata (volendo) per la challenge tls-alpn-01

**Reply by katanza12** [reply]
E allora non capisco cosa ci sia che on va...
